server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropSetTool.java

```diff
@@ -0,0 +1,391 @@
+
+    var siteConfig = opts.getSiteConfiguration();
+    try (ServerContext context = new ServerContext(siteConfig)) {
+
+      PropStoreKey<?> propKey = getPropKey(iid, opts, zrw);
+      switch (opts.getCmdMode()) {
+        case SET:
+          setProperty(context, propKey, opts);
+          break;
+        case DELETE:
+          deleteProperty(context, propKey, readPropNode(propKey, zrw), opts);
+          break;
+        case PRINT:
+          printProperties(context, propKey, readPropNode(propKey, zrw), opts);
+          break;
+        case ERROR:
+        default:
+          throw new IllegalArgumentException("Invalid operation requested");
+      }
+    }
+  }
+
+  private void setProperty(final ServerContext context, final PropStoreKey<?> propKey,
+      final Opts opts) {
+    LOG.trace("set {}", propKey);
+
+    if (!opts.setOpt.contains("=")) {
+      throw new IllegalArgumentException(
+          "Invalid set property format. Requires name=value, received " + opts.setOpt);
+    }
+    String[] tokens = opts.setOpt.split("=");
+    Map<String,String> propValue = Map.of(tokens[0].trim(), tokens[1].trim());
+    PropUtil.setProperties(context, propKey, propValue);
+  }
+
+  private void deleteProperty(final ServerContext context, final PropStoreKey<?> propKey,
+      VersionedProperties versionedProperties, final Opts opts) {
+    LOG.trace("delete {} - {}", propKey, opts.deleteOpt);
+    String p = opts.deleteOpt.trim();
+    if (p.isEmpty() || !Property.isValidPropertyKey(p)) {
+      throw new IllegalArgumentException("Invalid property name, Received: '" + p + "'");
+    }
+    // warn, but not throwing an error. If this was run in a script, allow the script to continue.
+    if (!versionedProperties.asMap().containsKey(p)) {
+      LOG.warn("skipping delete: property '{}' is not set for: {}- delete would have no effect", p,
+          propKey);
+      return;
+    }
+    PropUtil.removeProperties(context, propKey, List.of(p));
+  }
+
+  private void printProperties(final ServerContext context, final PropStoreKey<?> propKey,
+      final VersionedProperties props, final Opts opts) {
+    LOG.trace("print {}", propKey);
+
+    OutputStream outStream = System.out;
+
+    String scope;
+    if (propKey instanceof SystemPropKey) {
+      scope = "SYSTEM";
+    } else if (propKey instanceof NamespacePropKey) {
+      scope = "NAMESPACE";
+    } else if (propKey instanceof TablePropKey) {
+      scope = "TABLE";
+    } else {
+      scope = "unknown";
+    }
+
+    try (PrintWriter writer =
+        new PrintWriter(new BufferedWriter(new OutputStreamWriter(outStream, UTF_8)))) {
+      // header
+      writer.printf("- Instance name: %s\n", context.getInstanceName());
+      writer.printf("- Instance id: %s\n", context.getInstanceID());
+      writer.printf("- Property scope: - %s\n", scope);
+      writer.printf("- id: %s, data version: %d, timestamp: %s\n", propKey.getId(),
+          props.getDataVersion(), props.getTimestampISO());
```

Comment:
There should be a quiet mode to suppress this. Can also consider putting it in STDERR. So people can pipe the output and not catch all this, which is useful for listing props and grepping.

Reply:
The `-` for the header info was to support grep -v to strip those if desired.

Reply:
Okay, that makes sense, but grepping out a dash is unfriendly. You can't do `grep -v -`, you have to do `grep -v '[-]'` or `grep -v -- -`
Maybe instead, just `log.info()` this stuff, so it stands out and can be suppressed any number of ways, based on the user's logging config?

Reply:
Changes to use `+` to make 'grep -v` more user friendly